src/braket/aws/aws_session.py

```diff
@@ -825,3 +825,27 @@ def copy_session(
         # Preserve user_agent information
         copied_session._braket_user_agents = self._braket_user_agents
         return copied_session
+
+    def get_full_image_tag(self, image_uri: str) -> str:
+        """
+        Get verbose image tag from image uri.
+
+        Args:
+            image_uri (str): Image uri to get tag for.
+
+        Returns:
+            str: Verbose image tag for given image.
+        """
+        registry = image_uri.split(".")[0]
+        repository, tag = image_uri.split("/")[-1].split(":")
+        digest = self.ecr_client.batch_get_image(
+            registryId=registry,
+            repositoryName=repository,
+            imageIds=[{"imageTag": tag}],
+        )["images"][0]["imageId"]["imageDigest"]
+        tag = self.ecr_client.batch_get_image(
+            registryId=registry,
+            repositoryName=repository,
+            imageIds=[{"imageDigest": digest}],
+        )["images"][0]["imageId"]["imageTag"]
```

Comment:
Is the 0 here that you're looking at the first image that's returned? Is the order guaranteed (by date?) or is it random?

Reply:
Since we're only giving one digest here, we should only get a single image in the response (assuming no sha256 collisions lol). I believe the same holds true with a single tag pointing to a unique image

Reply:
Actually, learning more about how this works, I'll give a better answer shortly